src/Google.Pubsub.V1/SimplePubsub.cs

```diff
@@ -0,0 +1,215 @@
+        {
+            string topicName = GetTopicName(ProjectId, topicId);
+            const int RetryCount = 3;
+            for (int i = 0; i < RetryCount; i++)
+            {
+                try
+                {
+                    // Just check existence...
+                    await Publisher.GetTopicAsync(topicName);
+                    return new SimpleTopic(this, topicId);
+                }
+                catch (RpcException e) when (e.Status.StatusCode == StatusCode.NotFound)
+                {
+                    try
+                    {
+                        await Publisher.CreateTopicAsync(topicName);
+                        return new SimpleTopic(this, topicId);
+                    }
+                    catch (RpcException creationException) when (i < RetryCount - 1 && creationException.Status.StatusCode == StatusCode.AlreadyExists)
+                    {
+                        // Probably a race condition... someone else tried to create it just when we did.
+                        // Go back through the loop and check we can get it this time, failing after this happens 3 times.
+                    }
+                }
+            }
+            throw new InvalidOperationException("Bug in SimplePubsub: escaped the retry loop");
+        }
+
+        public SimpleSubscription GetSubscription(string subscriptionId)
+        {
+            var subscription = Subscriber.GetSubscription(GetSubscriptionName(ProjectId, subscriptionId));
+            return new SimpleSubscription(this,
+                SubscriptionTemplate.ParseName(subscription.Name)[1],
+                TopicTemplate.ParseName(subscription.Topic)[1]);
+        }
+
+        public async Task<SimpleSubscription> GetSubscriptionAsync(string subscriptionId)
+        {
+            var subscription = await Subscriber.GetSubscriptionAsync(SubscriptionTemplate.Expand(ProjectId, subscriptionId));
+            return new SimpleSubscription(this,
+                SubscriptionTemplate.ParseName(subscription.Name)[1],
+                TopicTemplate.ParseName(subscription.Topic)[1]);
+        }
+
+        public SimpleSubscription GetOrCreateSubscription(string subscriptionId, string topicId, int ackDeadlineSeconds = 60)
+        {
+            string subscriptionName = GetSubscriptionName(ProjectId, subscriptionId);
+            string topicName = GetTopicName(ProjectId, topicId);
+            const int RetryCount = 3;
+            for (int i = 0; i < RetryCount; i++)
+            {
+                try
+                {
+                    var subscription = Subscriber.GetSubscription(subscriptionName);
+                    if (subscription.Topic != topicName)
+                    {
+                        // TODO: Is this the most appropriate exception type?
+                        throw new InvalidOperationException($"Subscription {subscriptionName} is subscribed to topic {subscription.Topic}; expected topic was {topicName}");
```

Comment:
`InvalidOperationException` is the most accurate *existing* exception, but I think a new custom one makes sense.

But the message is actionable in terms of "It looks like your subscriptions are dodgy". It's sort of the equivalent of trying to open a file and finding it's not there.

Note that we're checking the subscribed topic, not the subscription ID - this isn't really an invariant.

Example of where this would be thrown:

```csharp
pubsub.GetOrCreateSubscription("foo", "bar");
pubsub.GetOrCreateSubscription("foo", "other");
```